git segment: unusual handling of port in ssh remote

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

I want to use the git segment to produce a link to the upstream. My upstream is of the form `ssh://git@host:222/path/to/repo.git`, which is translating into a url of `https://host/222/path/to/repo` - it's including the port as a URL segment for some reason.

The code looks like it's almost intended to handle this: [cleanUpstreamURL](https://github.com/JanDeDobbeleer/oh-my-posh/blob/254f0c4b64c6a73d7fbb3acc803ea49d8365dfb9/src/segments/git.go#L652) has a case to handle ssh with a specified port, but it seems to be hardcoded to ports that are exactly 4 characters long for no clear reason. I would guess changing the regex to match 1 to 5 digits instead of 4 exactly would fix this.

### Theme

Minimum working example:
```
blocks:
  - type: prompt
    alignment: left
    segments:
      - type: git
        style: plain
        options:
          fetch_upstream_icon: true
        templates:
          - "{{ url .UpstreamIcon .UpstreamURL }}"
```

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 28.9.0

Shell: shell (7.5.4)

Prompt:

]8;;https://forgejo.gorfey.cloud/222/luke/ansible-homelab\ ]8;;\

Segments:

ConsoleTitle(false)                       -   0 ms
Git(true)                                 -  78 ms

Run duration: 81.742ms

Cache path: C:\Users\gorfe\AppData\Local\Packages\ohmyposh.cli_96v55e8n804z4\LocalCache\Local\oh-my-posh

Config path: C:\LukeSource\personal-scripts\config\oh-my-posh-mwe.yaml

Logs:

[DEBUG] 12:45:26.299 debug.go:39 → logging enabled, raw mode: true
[DEBUG] 12:45:26.300 store.go:]:141 → (session) key not found: prompt_count_cache
[TRACE] 12:45:26.300 store.go:](session prompt_count_cache) - 0s
[TRACE] 12:45:26.300 terminal.go:setPromptCount() - 0s
[TRACE] 12:45:26.300 terminal.go:GOOS() - 0s
[DEBUG] 12:45:26.300 terminal.go:setPwd:102 → C:\LukeSource\ansible-homelab
[TRACE] 12:45:26.300 terminal.go:setPwd() - 0s
[TRACE] 12:45:26.300 terminal.go:Init() - 0s
[DEBUG] 12:45:26.300 path_windows.go:PackageFamilyName:49 → PackageFamilyName: ohmyposh.cli_96v55e8n804z4
[TRACE] 12:45:26.300 path_windows.go:PackageFamilyName() - 0s
[TRACE] 12:45:26.300 path.go:Path() - 0s
[DEBUG] 12:45:26.301 store.go:init:94 → (device) loading reload
[DEBUG] 12:45:26.301 store.go:init:94 → (device) loading INITVERSIONPWSH
[DEBUG] 12:45:26.301 store.go:init:94 → (device) loading DSC_SHELL
[DEBUG] 12:45:26.301 store.go:init:94 → (device) loading DSC_CONFIGURATION
[TRACE] 12:45:26.301 store.go:init(device omp.cache) - 505.4µs
[TRACE] 12:45:26.301 init.go:SessionID() - 0s
[TRACE] 12:45:26.301 path.go:Path() - 0s
[DEBUG] 12:45:26.301 store.go:init:94 → (session) loading CONFIG
[DEBUG] 12:45:26.301 store.go:init:94 → (session) loading CONFIG_SOURCE
[DEBUG] 12:45:26.301 store.go:init:94 → (session) loading toggle_cache
[DEBUG] 12:45:26.301 store.go:init:94 → (session) loading prompt_count_cache
[TRACE] 12:45:26.301 store.go:init(session pwsh.46e0ce5b-2200-4aa7-aef4-0c0d84a27ac9.omp.cache) - 0s
[DEBUG] 12:45:26.301 store.go:]:154 → (device) found entry: reload - true
[TRACE] 12:45:26.301 store.go:](device reload) - 0s
[DEBUG] 12:45:26.301 gob.go:Get:29 → reload mode enabled
[DEBUG] 12:45:26.301 store.go:]:154 → (session) found entry: CONFIG_SOURCE - C:\LukeSource\personal-scripts\config\oh-my-posh-mwe.yaml
[TRACE] 12:45:26.301 store.go:](session CONFIG_SOURCE) - 0s
[DEBUG] 12:45:26.301 load.go:isTheme:384 → C:\LukeSource\personal-scripts\config\oh-my-posh-mwe.yaml is not a theme
[TRACE] 12:45:26.301 load.go:resolveConfigLocation() - 0s
[DEBUG] 12:45:26.301 store.go:]:154 → (device) found entry: DSC_CONFIGURATION - [0xc000316000 0xc000316180]
[TRACE] 12:45:26.301 store.go:](device DSC_CONFIGURATION) - 0s
[DEBUG] 12:45:26.301 home.go:12 → C:\Users\gorfe
[DEBUG] 12:45:26.301 resource.go:Add:42 → item already exists
[TRACE] 12:45:26.301 load.go:read() - 507.4µs
[DEBUG] 12:45:26.301 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 12:45:26.301 store.go:](session toggle_cache) - 0s
[DEBUG] 12:45:26.301 store.go:]:177 → (session) setting entry: toggle_cache - map[] with duration: infinite
[TRACE] 12:45:26.301 store.go:](session toggle_cache) - 0s
[DEBUG] 12:45:26.301 store.go:]:177 → (device) setting entry: DSC_CONFIGURATION - [0xc000316000 0xc000316180] with duration: infinite
[TRACE] 12:45:26.301 store.go:](device DSC_CONFIGURATION) - 0s
[TRACE] 12:45:26.301 load.go:Parse() - 507.4µs
[TRACE] 12:45:26.301 load.go:Load() - 507.4µs
[DEBUG] 12:45:26.301 store.go:]:177 → (session) setting entry: CONFIG_SOURCE - C:\LukeSource\personal-scripts\config\oh-my-posh-mwe.yaml with duration: infinite
[TRACE] 12:45:26.301 store.go:](session CONFIG_SOURCE) - 0s
[TRACE] 12:45:26.301 gob.go:Base64() - 0s
[DEBUG] 12:45:26.301 store.go:]:177 → (session) setting entry: CONFIG - /gH5fwMBAQZDb25maWcB/4AAAR8BB1BhbGV0dGUB/4IAAQtEZWJ1Z1Byb21wdAH/hAABA1ZhcgH/jgABCFBhbGV0dGVzAf+QAAEJVmFsaWRMaW5lAf+EAAEPU2Vjb25kYXJ5UHJvbXB0Af+EAAEPVHJhbnNpZW50UHJvbXB0Af+EAAEJRXJyb3JMaW5lAf+EAAEETWFwcwH/lAABB1VwZ3JhZGUB/5gAAQdFeHRlbmRzAQwAAQtBY2NlbnRDb2xvcgEMAAEUQ29uc29sZVRpdGxlVGVtcGxhdGUBDAABA1BXRAEMAAEGU291cmNlAQwAAQZGb3JtYXQBDAABElRlcm1pbmFsQmFja2dyb3VuZAEMAAEOVG9vbFRpcHNBY3Rpb24BDAABBkJsb2NrcwH/ngABBUN5Y2xlAf+iAAENSVRlcm1GZWF0dXJlcwH/pAABCFRvb2x0aXBzAf+cAAEHVmVyc2lvbgEEAAENTWlncmF0ZUdseXBocwECAAEFQXN5bmMBAgABEFNoZWxsSW50ZWdyYXRpb24BAgABCkZpbmFsU3BhY2UBAgABDVVwZ3JhZGVOb3RpY2UBAgABDlBhdGNoUHdzaEJsZWVkAQIAAQtBdXRvVXBncmFkZQECAAEXRW5hYmxlQ3Vyc29yUG9zaXRpb25pbmcBAgAAABf/gQQBAQdQYWxldHRlAf+CAAEMAQwAAP4CFf+DAwEBB1NlZ21lbnQB/4QAASIBB09wdGlvbnMB/4YAAQpQcm9wZXJ0aWVzAf+GAAEFQ2FjaGUB/4gAAQVBbGlhcwEMAAEOTGVhZGluZ0RpYW1vbmQBDAABD1RyYWlsaW5nRGlhbW9uZAEMAAEIVGVtcGxhdGUBDAABCkZvcmVncm91bmQBDAABDlRlbXBsYXRlc0xvZ2ljAQwAAQ9Qb3dlcmxpbmVTeW1ib2wBDAABCkJhY2tncm91bmQBDAABBkZpbGxlcgEMAAEEVHlwZQEMAAEFU3R5bGUBDAABFkxlYWRpbmdQb3dlcmxpbmVTeW1ib2wBDAABE0ZvcmVncm91bmRUZW1wbGF0ZXMB/4oAAQRUaXBzAf+MAAETQmFja2dyb3VuZFRlbXBsYXRlcwH/igABCVRlbXBsYXRlcwH/igABDkV4Y2x1ZGVGb2xkZXJzAf+MAAEOSW5jbHVkZUZvbGRlcnMB/4wAAQVOZWVkcwH/jAABB1RpbWVvdXQBBAABCE1heFdpZHRoAQQAAQhNaW5XaWR0aAEEAAEIRHVyYXRpb24BBAABCk5hbWVMZW5ndGgBBAABBUluZGV4AQQAAQtJbnRlcmFjdGl2ZQECAAEHRW5hYmxlZAECAAEHTmV3bGluZQECAAEPSW52ZXJ0UG93ZXJsaW5lAQIAAQVGb3JjZQECAAEHVG9nZ2xlZAECAAAAE/+FBAEBA01hcAH/hgABDAEQAAAt/4cDAQEFQ2FjaGUB/4gAAQIBCER1cmF0aW9uAQwAAQhTdHJhdGVneQEMAAAAEv+JAgEBBExpc3QB/4oAAQwAABb/iwIBAQhbXXN0cmluZwH/jAABDAAAJ/+NBAEBF21hcFtzdHJpbmddaW50ZXJmYWNlIHt9Af+OAAEMARAAAC3/jwMBAQhQYWxldHRlcwH/kAABAgEETGlzdAH/kgABCFRlbXBsYXRlAQwAAAAp/5EEAQEYbWFwW3N0cmluZ11jb2xvci5QYWxldHRlAf+SAAEMAf+CAAA//5MDAQEGQ29uZmlnAf+UAAEDAQhVc2VyTmFtZQH/lgABCEhvc3ROYW1lAf+WAAEJU2hlbGxOYW1lAf+WAAAAE/+VBAEBA01hcAH/lgABDAEMAABc/5cDAQEGQ29uZmlnAf+YAAEGAQZTb3VyY2UBDAABCEludGVydmFsAQwAAQZMYXRlc3QBDAABBEF1dG8BAgABDURpc3BsYXlOb3RpY2UBAgABBUZvcmNlAQIAAAAe/50CAQEPW10qY29uZmlnLkJsb2NrAf+eAAH/mgAA/6r/mQMBAv+aAAEMAQRUeXBlAQwAAQlBbGlnbm1lbnQBDAABBkZpbGxlcgEMAAEIT3ZlcmZsb3cBDAABDkxlYWRpbmdEaWFtb25kAQwAAQ9UcmFpbGluZ0RpYW1vbmQBDAABCFNlZ21lbnRzAf+cAAEITWF4V2lkdGgBBAABCE1pbldpZHRoAQQAAQdOZXdsaW5lAQIAAQVGb3JjZQECAAEFSW5kZXgBBAAAACD/mwIBARFbXSpjb25maWcuU2VnbWVudAH/nAAB/4QAABT/oQIBAQVDeWNsZQH/ogAB/6AAACr/nwMBAv+gAAECAQpCYWNrZ3JvdW5kAQwAAQpGb3JlZ3JvdW5kAQwAAAAb/6MCAQENSVRlcm1GZWF0dXJlcwH/pAABDAAA/7f/gAoBA2NkbgEEMTY4aAAFOUM6XEx1a2VTb3VyY2VccGVyc29uYWwtc2NyaXB0c1xjb25maWdcb2gtbXktcG9zaC1td2UueWFtbAEEeWFtbAMBAQZwcm9tcHQBBGxlZnQFAQEBE2ZldGNoX3Vwc3RyZWFtX2ljb24EYm9vbAICAAEMA2dpdAEFcGxhaW4FASR7eyB1cmwgLlVwc3RyZWFtSWNvbiAuVXBzdHJlYW1VUkwgfX0AAAA= with duration: infinite
[TRACE] 12:45:26.301 store.go:](session CONFIG) - 0s
[TRACE] 12:45:26.301 gob.go:Store() - 0s
[TRACE] 12:45:26.301 gob.go:Get() - 507.4µs
[TRACE] 12:45:26.301 terminal.go:Shell() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[DEBUG] 12:45:26.301 store.go:]:141 → (session) key not found: template_cache
[TRACE] 12:45:26.301 store.go:](session template_cache) - 0s
[TRACE] 12:45:26.301 cache.go:restoreCache() - 0s
[TRACE] 12:45:26.301 terminal_windows.go:Root() - 0s
[TRACE] 12:45:26.301 terminal.go:Shell() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[TRACE] 12:45:26.301 terminal.go:StatusCodes() - 0s
[TRACE] 12:45:26.301 terminal_windows.go:IsWsl() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[DEBUG] 12:45:26.301 home.go:12 → C:\Users\gorfe
[TRACE] 12:45:26.301 terminal_windows.go:IsWsl() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[TRACE] 12:45:26.301 terminal.go:GOOS() - 0s
[DEBUG] 12:45:26.301 terminal.go:User:236 → gorfe
[TRACE] 12:45:26.301 terminal.go:User() - 0s
[DEBUG] 12:45:26.301 terminal.go:Host:253 → Luke-PC
[TRACE] 12:45:26.301 terminal.go:Host() - 0s
[TRACE] 12:45:26.301 terminal.go:GOOS() - 0s
[DEBUG] 12:45:26.301 terminal.go:Getenv:64 → NO DATA
[TRACE] 12:45:26.301 terminal.go:Getenv(SHLVL) - 0s
[DEBUG] 12:45:26.301 writer.go:Init:107 → terminal program: Windows Terminal
[DEBUG] 12:45:26.301 writer.go:Init:108 → terminal shell: shell
[DEBUG] 12:45:26.301 terminal.go:Getenv:64 → NO DATA
[TRACE] 12:45:26.301 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[DEBUG] 12:45:26.301 store.go:]:141 → (device) key not found: accent_color
[TRACE] 12:45:26.301 store.go:](device accent_color) - 0s
[DEBUG] 12:45:26.301 terminal_windows.go:WindowsRegistryKeyValue:205 → ColorizationColor(DWORD): 0xC4409EFE
[TRACE] 12:45:26.301 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 12:45:26.301 colors_windows.go:GetAccentColor() - 0s
[TRACE] 12:45:26.301 colors.go:SetAccentColor() - 0s
[TRACE] 12:45:26.301 terminal.go:Shell() - 0s
[DEBUG] 12:45:26.301 terminal.go:Getenv:64 → 7.5.4
[TRACE] 12:45:26.301 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 12:45:26.301 debug.go:PrintDebug:24 → segment: Title
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[DEBUG] 12:45:26.301 terminal.go:Getenv:64 → 40
[TRACE] 12:45:26.301 terminal.go:Getenv(POSH_CURSOR_LINE) - 0s
[DEBUG] 12:45:26.301 terminal.go:Getenv:64 → 1
[TRACE] 12:45:26.301 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[TRACE] 12:45:26.301 terminal.go:Flags() - 0s
[DEBUG] 12:45:26.301 home.go:12 → C:\Users\gorfe
[TRACE] 12:45:26.301 terminal.go:GOOS() - 0s
[DEBUG] 12:45:26.301 segment.go:Execute:172 → segment: Git
[DEBUG] 12:45:26.301 store.go:]:154 → (session) found entry: toggle_cache - map[]
[TRACE] 12:45:26.301 store.go:](session toggle_cache) - 0s
[DEBUG] 12:45:26.301 segment.go:isToggled:293 → no toggles found
[DEBUG] 12:45:26.301 map.go:Bool:112 → disable_with_jj: false
[DEBUG] 12:45:26.302 terminal.go:ResolveSymlink:195 → C:\LukeSource\ansible-homelab
[TRACE] 12:45:26.302 terminal.go:ResolveSymlink(C:\LukeSource\ansible-homelab) - 534.5µs
[TRACE] 12:45:26.302 terminal.go:HasParentFilePath(.git) - 534.5µs
[DEBUG] 12:45:26.302 map.go:Bool:112 → fetch_bare_info: false
[TRACE] 12:45:26.302 terminal.go:GOOS() - 0s
[DEBUG] 12:45:26.327 terminal.go:CommandPath:304 → C:\Program Files\Git\cmd\git.exe
[TRACE] 12:45:26.327 terminal.go:CommandPath(git.exe) - 25.4893ms
[TRACE] 12:45:26.327 terminal.go:HasCommand(git.exe) - 25.4893ms
[DEBUG] 12:45:26.327 home.go:12 → C:\Users\gorfe
[TRACE] 12:45:26.327 terminal.go:GOOS() - 0s
[TRACE] 12:45:26.327 terminal.go:GOOS() - 0s
[DEBUG] 12:45:26.327 map.go:Bool:112 → fetch_user: false
[DEBUG] 12:45:26.327 map.go:String:77 → source: cli
[DEBUG] 12:45:26.327 map.go:Bool:112 → fetch_status: false
[DEBUG] 12:45:26.327 terminal.go:FileContent:212 ↓
    88600201aec19155cd32a3683262f865cddaea16
    
[TRACE] 12:45:26.327 terminal.go:FileContent(C:\LukeSource\ansible-homelab\.git/HEAD) - 0s
[DEBUG] 12:45:26.327 git.go:updateHEADReference:978 → HEADRef: 88600201aec19155cd32a3683262f865cddaea16
[DEBUG] 12:45:26.354 terminal.go:RunCommand:280 → 88600201aec19155cd32a3683262f865cddaea16
[TRACE] 12:45:26.354 terminal.go:RunCommand(git.exe -C C:/LukeSource/ansible-homelab --no-optional-locks -c core.quotepath=false -c color.status=false rev-parse HEAD) - 27.3606ms
[ERROR] 12:45:26.380 terminal.go:RunCommand:277 → exit status 128
[DEBUG] 12:45:26.380 terminal.go:RunCommand:280 ↓
    fatal: No names found, cannot describe anything.
    
[TRACE] 12:45:26.380 terminal.go:RunCommand(git.exe -C C:/LukeSource/ansible-homelab --no-optional-locks -c core.quotepath=false -c color.status=false describe --tags --exact-match) - 25.5596ms
[DEBUG] 12:45:26.380 map.go:String:77 → commit_icon: 
[DEBUG] 12:45:26.380 map.go:Bool:120 → fetch_upstream_icon: true
[DEBUG] 12:45:26.380 map.go:String:77 → git_icon:  
[DEBUG] 12:45:26.380 terminal.go:FileContent:212 ↓
    [core]
    	repositoryformatversion = 0
    	filemode = false
    	bare = false
    	logallrefupdates = true
    	symlinks = false
    	ignorecase = true
    [remote "origin"]
    	url = ssh://git@forgejo.gorfey.cloud:222/luke/ansible-homelab.git
    	fetch = +refs/heads/*:refs/remotes/origin/*
    [branch "main"]
    	remote = origin
    	merge = refs/heads/main
    	rebase = true
    	vscode-merge-base = origin/main
    [branch "mytempbranch"]
    	remote = origin
    	merge = refs/heads/main
    	rebase = true
    
[TRACE] 12:45:26.380 terminal.go:FileContent(C:\LukeSource\ansible-homelab\.git/config) - 0s
[DEBUG] 12:45:26.380 git.go:getRemoteURL:1075 → remote url found in config: ssh://git@forgejo.gorfey.cloud:222/luke/ansible-homelab.git
[DEBUG] 12:45:26.380 map.go:KeyValueMap:187 → upstream_icons: map[]
[DEBUG] 12:45:26.380 text.go:patchTemplate:149 → {{ url .Data.UpstreamIcon .Data.UpstreamURL }}
[TRACE] 12:45:26.381 terminal.go:Shell() - 0s
[TRACE] 12:45:26.381 terminal.go:Shell() - 0s
[TRACE] 12:45:26.381 path.go:Path() - 0s
```

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -650,7 +650,7 @@ func (g *Git) cleanUpstreamURL(url string) string {
 	}
 
 	// ssh://user@host.xz:1234/path/to/repo.git/
-	match = regex.FindNamedRegexMatch(`(ssh|ftp|git|rsync)://(.*@)?(?P<URL>[a-z0-9.-]+)(:[0-9]{4})?/(?P<PATH>.*).git`, url)
+	match = regex.FindNamedRegexMatch(`(ssh|ftp|git|rsync)://(.*@)?(?P<URL>[a-z0-9.-]+)(:[0-9]{1,5})?/(?P<PATH>.*).git`, url)
 	if len(match) == 0 {
 		// host.xz:/path/to/repo.git/
 		match = regex.FindNamedRegexMatch(`^(?P<URL>[a-z0-9.-]+):(?P<PATH>[\w.\-~/@]+)$`, url)

--- a/src/segments/git_test.go
+++ b/src/segments/git_test.go
@@ -651,6 +651,7 @@ func TestGitCleanSSHURL(t *testing.T) {
 		{Case: "regular URL", Expected: "https://src.example.com/user/repo", Upstream: "/src.example.com/user/repo.git"},
 		{Case: "domain:path", Expected: "https://host.xz/path/to/repo", Upstream: "host.xz:/path/to/repo.git/"},
 		{Case: "ssh with port", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz:1234/path/to/repo.git"},
+		{Case: "ssh with 3-digit port", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz:234/path/to/repo.git"},
 		{Case: "ssh with port, trailing slash", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz:1234/path/to/repo.git/"},
 		{Case: "ssh without port", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://user@host.xz/path/to/repo.git/"},
 		{Case: "ssh port, no user", Expected: "https://host.xz/path/to/repo", Upstream: "ssh://host.xz:1234/path/to/repo.git"},

